### ISSUE TITLE
Make predictions for classfication task of infiniteBoost

### ISSUE THREAD

**Issue by vatn** [arogozhnikov/infiniteboost#5]
I am trying to test infiniteboost with titanic dataset from kaggle.

titanic_df = pd.read_csv("train_cleaned")
y = titanic_df["Survived"].values
X = titanic_df.drop("Survived", axis = 1).values
clf = InfiniteBoosting(loss = LogisticLoss(),n_estimators= 100)
X = BinTransformer().fit_transform(X)
clf.fit(X,y)
ypred = clf.staged_decision_function(X)
y_last_pred = clf.decision_function(X)
y_last_pred

It is a classification problem, how can I know the infiniteBoost will consider it as a classification problem?(the target variable is y, the value of y is 0 or 1(int)).
And when I used the decision_function to make predictions, I found it doesn't look like neither  probabilities nor classes.
So, how does inifiniteBoost work with classification tasks and how to use it to make predictions of probabilities?

**Comment by arogozhnikov** [comment]
Hi, 
> It is a classification problem, how can I know the infiniteBoost will consider it as a classification problem?(the target variable is y, the value of y is 0 or 1(int)).

You've provided this information by setting loss function to LogLoss (usual classification loss function).

> And when I used the decision_function to make predictions, I found it doesn't look like neither probabilities nor classes.

Decision function is raw output of boosting, which shall be converted to probabilities (or something else) depending on the loss function used in training. In your case (LogLoss):

```python
from scipy.special import expit
p_1 = expit(2 * clf.decision_function(X)) # probability to belong to class 1
p_0 = 1 - p_1 # probability to belong to class 0
```

**Comment by vatn** [comment]
Thank you !
